controllers/pipelinerun_controller.go

```diff
@@ -0,0 +1,205 @@
+	} else if tekton.IsReleasePipelineRun(pipelineRun) {
+		return r.updateReleaseStatus(ctx, pipelineRun)
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// createRelease creates a new Release by using the information provided in the Build PipelineRun.
+func (r *PipelineRunReconciler) createRelease(ctx context.Context, pipelineRun *tektonv1beta1.PipelineRun) (ctrl.Result, error) {
+	log := r.Log.WithValues()
+
+	log.Info("Creating new Release",
+		"PipelineRun.Name", pipelineRun.Name, "PipelineRun.Namespace", pipelineRun.Namespace)
+
+	component, err := r.getComponent(ctx, pipelineRun)
+	if err != nil {
+		log.Error(err, "Failed to get Component",
+			"PipelineRun.Name", pipelineRun.Name, "PipelineRun.Namespace", pipelineRun.Namespace)
+
+		return ctrl.Result{}, nil
+	}
+
+	releaseLink, err := r.getReleaseLink(ctx, component)
+	if err != nil {
+		log.Error(err, "Failed to get ReleaseLink",
+			"Component.Name", component.Name, "Component.Namespace", component.Namespace)
+
+		return ctrl.Result{}, nil
+	}
+
+	release := v1alpha1.NewRelease(component, releaseLink)
+	err = r.Create(ctx, release)
+	if err != nil {
+		log.Error(err, "Failed to create Release")
+
+		return ctrl.Result{}, err
+	}
+
+	release.Status.Trigger = "automated"
+
+	return helpers.UpdateStatus(r.Client, ctx, release)
+}
+
+// updateReleaseStatus updates the status of the Release referenced in a Release PipelineRun. The new state will
+// reflect the state of the execution of the Release Pipeline.
+func (r *PipelineRunReconciler) updateReleaseStatus(ctx context.Context, pipelineRun *tektonv1beta1.PipelineRun) (ctrl.Result, error) {
+	log := r.Log.WithValues()
+
+	release, err := r.getRelease(ctx, pipelineRun)
+	if err != nil {
+		log.Error(err, "Failed to update Release")
+		return ctrl.Result{}, nil
+	}
+
+	// This log could be too chatty as the PipelineRun gets frequently updated.
+	log.V(2).Info("Updating existing Release",
+		"Release.Name", release.Name, "Release.Namespace", release.Namespace)
+
+	if pipelineRun.IsDone() {
+		condition := pipelineRun.Status.GetCondition(apis.ConditionSucceeded)
+		if condition.IsTrue() {
+			release.Status.SetSucceededCondition()
+		} else {
+			release.Status.SetFailedCondition(pipelineRun)
+		}
+	} else if release.Status.ReleasePipelineRun == "" {
+		release.Status.SetRunningCondition(pipelineRun)
+	}
+
+	return helpers.UpdateStatus(r.Client, ctx, release)
+}
+
+// getComponent loads from the cluster the Component referenced in the given PipelineRun. If the PipelineRun doesn't specify
+// a Component or this is not found in the cluster, an error will be returned.
+func (r *PipelineRunReconciler) getComponent(ctx context.Context, pipelineRun *tektonv1beta1.PipelineRun) (*hasv1alpha1.Component, error) {
+	if componentName, found := pipelineRun.Labels["build.appstudio.openshift.io/component"]; found {
+		component := &hasv1alpha1.Component{}
+		err := r.Get(ctx, types.NamespacedName{
+			Namespace: pipelineRun.Namespace,
```

Comment:
so the pipeline runs will all be in the namespace of the component, not in some "release-service namespace"

I know I heard about some release-service namespace in meetings a while ago but no idea if that was dropped or not

Reply:
Build ones in the component namespace. Ours in the target workspace.

Reply:
ah, this function is only used in the build pipeline tree of logic. Got it, thanks